docs/csharp/pattern-matching.md

```diff
@@ -1,268 +1,70 @@
 
-[!code-csharp[ClassicIsExpression](../../samples/snippets/csharp/PatternMatching/GeometricUtilities.cs#02_ClassicIsExpression "Classic type pattern using is")]
+:::code language="csharp" source="snippets/patterns/Program.cs" ID="TypeCheckDisposable":::
 
-That code above is a classic expression of the *type pattern*: You're testing a variable
-to determine its type and taking a different action based on that type.
+The same tests can be applied in a `switch` expression to test a variable against multiple different types. You can use that information to create better algorithms based on the specific runtime type.
 
-This code becomes simpler using extensions to the `is` expression to assign
-a variable if the test succeeds:
+## Compare discrete values
 
-[!code-csharp[IsPatternExpression](../../samples/snippets/csharp/PatternMatching/GeometricUtilities.cs#03_IsPatternExpression "is pattern expression")]
+You can also test a variable to find a match on specific values. The following code shows one example where you test a value against all possible values declared in an enumeration:
 
-In this updated version, the `is` expression both tests the variable and assigns
-it to a new variable of the proper type. Also, notice that this version includes
-the `Rectangle` type, which is a `struct`. The new `is` expression works with
-value types as well as reference types.
+:::code language="csharp" source="snippets/patterns/Simulation.cs" ID="PerformOperation":::
 
-Language rules for pattern matching expressions help you avoid misusing
-the results of a match expression. In the example above, the variables `s`,
- `c`, and `r` are only in scope and definitely assigned when the respective
-pattern match expressions have `true` results. If you try to use either
-variable in another location, your code generates compiler errors.
+The previous example demonstrates a method dispatch based on the value of an enumeration. The final `_` case is a [*discard pattern*](language-reference/operators/patterns.md#discard-pattern) that matches all values. It handles any error conditions where the value doesn't match one of the defined `enum` values. If you omit that switch arm, the compiler warns that you haven't handled all possible input values. At runtime, the `switch` expression throws an exception if the object being examined doesn't match any of the switch arms. You could use numeric constants instead of a set of enum values. You can also use this similar technique for constant string values that represent the commands:
```

Comment:
```suggestion
The previous example demonstrates a method dispatch based on the value of an enumeration. The final `_` case is a [*discard pattern*](language-reference/operators/patterns.md#discard-pattern) that matches all values. It handles any error conditions where the value doesn't match one of the defined `enum` values. If you omit that switch arm, the compiler warns that you haven't handled all possible input values. At run time, the `switch` expression throws an exception if the object being examined doesn't match any of the switch arms. You could use numeric constants instead of a set of enum values. You can also use this similar technique for constant string values that represent the commands:
```